streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -783,12 +781,13 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
                 stateDirectory,
                 delegatingStateRestoreListener,
                 i + 1);
-            threadState.put(threads[i].getId(), threads[i].state());
-            storeProviders.add(new StreamThreadStateStoreProvider(threads[i]));
+            threads.add(i, streamThread);
+            threadState.put(streamThread.getId(), streamThread.state());
+            storeProviders.add(new StreamThreadStateStoreProvider(threads.get(i)));
```

Comment:
```suggestion
            storeProviders.add(new StreamThreadStateStoreProvider(streamThread));
```

`get(i)` is also O(n) for a linked list. Again, this is admittedly a nit, since the list is small.

Reply:
Good catch